rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -229,6 +230,30 @@ public RMStateFull getRMStateFull(String sessionId) throws NotConnectedException
         return orThrowRpe(RMStateCaching.getRMStateFull());
     }
 
+    @Override
+    public String getModelHosts() throws PermissionRestException {
+        RMStateFull state = orThrowRpe(RMStateCaching.getRMStateFull());
+        return "PA:LIST(," +
+               state.getNodesEvents()
+                    .stream()
+                    .map(event -> event.getHostName())
+                    .distinct()
+                    .collect(Collectors.joining(",")) +
+               ")";
```

Comment:
Is it normal that `PA:LIST(,` start with a comma ?
I think it would be more readable using String.format
```
return String.format("PA:LIST(%s)",
               state.getNodesEvents()
                    .stream()
                    .map(event -> event.getHostName())
                    .distinct()
                    .collect(Collectors.joining(","));

Reply:
ok for format.

Yes for PA:LIST it is normal (unfortunately). I need to produce a list which allows an empty choice. And there is no other way to do it than directly inside the endpoint.

Empty choice is needed because of a bootstrap issue. When someone wants to define a workflow or task variable with PA:MODEL_FROM_URL pointing to this endpoint. The variable must have a valid default value (otherwise the studio does not allow to create it), which thus must be a valid hostname or node source. This makes the workflow non-portable and it could not be stored in the catalog proactive-examples.